main.tf

```diff
@@ -174,31 +178,62 @@ resource "google_cloud_run_service" "api" {
       service_account_name = google_service_account.runsa.email
       containers {
         image = local.api_image
-        env {
-          name  = "redis_host"
-          value = google_redis_instance.main.host
+        dynamic "env" {
+            for_each = var.database_type == "postgres" ? ["redis_host"] : ["REDISHOST"]
+            content {
+              name  = env.value
+              value = google_redis_instance.main.host
+            }
         }
-        env {
-          name  = "db_host"
-          value = google_sql_database_instance.main.ip_address[0].ip_address
+        dynamic "env" {
+            for_each = var.database_type == "postgres" ? ["redis_port"] : ["REDISPORT"]
+            content {
+              name  = env.value
+              value = "6379"
+            }
         }
-        env {
-          name  = "db_user"
-          value = google_service_account.runsa.email
+        dynamic "env" {
+            for_each = var.database_type == "postgres" ? ["db_host"] : ["todo_host"]
+            content {
+              name  = env.value
+              value = google_sql_database_instance.main.ip_address[0].ip_address
+            }
         }
-        env {
-          name  = "db_conn"
-          value = google_sql_database_instance.main.connection_name
+        dynamic "env" {
+            for_each = var.database_type == "postgres" ? ["db_name"] : ["todo_name"]
+            content {
+              name  = env.value
+              value = "todo"
+            }
         }
-        env {
-          name  = "db_name"
-          value = "todo"
+        dynamic "env" {
+            for_each = var.database_type == "postgres" ? [1] : []
+            content {
+              name  = "db_user"
+              value = google_service_account.runsa.email
+            }
         }
-        env {
-          name  = "redis_port"
-          value = "6379"
+        dynamic "env" {
+            for_each = var.database_type == "postgres" ? [1] : []
+            content {
+              name  = "db_conn"
+              value = google_sql_database_instance.main.connection_name
+            }
+        }
+        dynamic "env" {
+            for_each = var.database_type == "mysql" ? [1] : []
+            content {
+              name  = "todo_user"
+              value = "foo"
+            }
+        }
+        dynamic "env" {
+            for_each = var.database_type == "mysql" ? [1] : []
+            content {
+              name  = "todo_pass"
+              value = "bar"
+            }
```

Comment:
**Suggestion:** Let's clean this up a little by introducing `api_env_vars_postgres` and `api_env_vars_mysql` locals:
```
locals {
...
  api_env_vars_postgres = {
    redis_host = google_redis_instance.main.host
    db_host    = google_sql_database_instance.main.ip_address[0].ip_address
    db_user    = google_service_account.runsa.email
    db_conn    = google_sql_database_instance.main.connection_name
    db_name    = "todo"
    redis_port = "6379"
  }

  api_env_vars_mysql = {
    REDISHOST  = google_redis_instance.main.host
    todo_host  = google_sql_database_instance.main.ip_address[0].ip_address
    todo_user  = "foo"               
    todo_pass  = "bar"
    todo_name  = "todo"
    REDISPORT  = "6379"
  }
}
```
and using something like:
```
resource "google_cloud_run_service" "api" {
...
        dynamic "env" {
          for_each = var.database_type == "postgres" ? local.api_env_vars_postgresql : local.api_env_vars_mysql
          content {
            name  = env.key
            value = env.value
          }
        } 
...
```

Rationale: I want to make it easier to later add more database integrations (e.g., AlloyDB, SQL Server on Cloud SQL, etc.).

Reply:
When I added this, I searched quite a bit on how to simplify this piece. Didn't like the previous code.

Added your suggestion. thanks!